retrievers/merger_retriever.go

```diff
@@ -0,0 +1,44 @@
+package retrievers
+
+import (
+	"context"
+
+	"github.com/tmc/langchaingo/callbacks"
+	"github.com/tmc/langchaingo/schema"
+)
+
+var _ schema.Retriever = &MergerRetriever{}
+
+// MergerRetriever is a retriever that merges the results of multiple retrievers.
+type MergerRetriever struct {
+	Retrievers       []schema.Retriever
+	CallbacksHandler callbacks.Handler
+}
+
+// NewMergerRetriever creates a new MergerRetriever.
+func NewMergerRetriever(
```

Comment:
maybe it's better to use MergeRetriever for the name

Reply:
We're consistent with the naming of langchain python...they call it Merger Retriever in doc and mergerretiever in code.